### ISSUE TITLE
Can't reproduce the result of "Benchmark 2: Fine-Tuning RoBERTa on GLUE tasks"

### ISSUE THREAD

**Issue by CrazyElements** [jiaweizzhao/GaLore#25]
Has anyone successfully replicated the results of fine-tuning tasks?
I followed the hyperparameters outlined in the REAMDE and the paper, and tried cola and mrpc tasks on a single GPU without gradient accumulation. However, the results I obtained differed from those reported in the paper.
And here are best performances of my runs
- mrpc: 0.8971(92.25)
- cola: 0.6274(0.6035),
where numbers in parentheses are the results of the paper.
I would appreciate any assistance from someone who can provide insights on this matter.

**Comment by jiaweizzhao** [comment]
I am happy to help @CrazyElements  
Can you provide the full training script and hyperparameters you are using?

**Comment by jiaweizzhao** [comment]
I tried and it works as expected. The issue might be we report F1 score of mrpc in the paper and causes the confusion. I will change it back to accuracy in the new revision.

**Comment by jiaweizzhao** [comment]
This might be due to the choice of the random seed. I did a quick sweep using my previous setup (based on the config you provided):
`python run_glue.py \
    --model_name_or_path roberta-base \
    --task_name mrpc \
    --enable_galore \
    --lora_all_modules \
    --max_length 512 \
    --seed=1234 \
    --lora_r 4 \
    --galore_scale 16 \
    --per_device_train_batch_size 32 \
    --update_proj_gap 500 \
    --learning_rate 2e-5 \
    --num_train_epochs 20 \
    --output_dir results/ft/roberta_base/mrpc`
This gives {"eval_accuracy": 0.8970588235294118, "eval_f1": 0.925531914893617}

**Comment by jiaweizzhao** [comment]
We use the avg score of repeated runs. We will release the fine-tuning scripts later, along with a few more fine-tuning experiments.

**Comment by nikitaved** [comment]
@jiaweizzhao , may I ask about the dtype used to report numbers in the paper? I assume it is `float`, right? `half` does not learn with these hyperparams, and `bfloat16` shows lower numbers...
